framework/src/org/apache/cordova/ICordovaCookieManager.java

```diff
@@ -0,0 +1,45 @@
+/*
+       Licensed to the Apache Software Foundation (ASF) under one
+       or more contributor license agreements.  See the NOTICE file
+       distributed with this work for additional information
+       regarding copyright ownership.  The ASF licenses this file
+       to you under the Apache License, Version 2.0 (the
+       "License"); you may not use this file except in compliance
+       with the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing,
+       software distributed under the License is distributed on an
+       "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+       KIND, either express or implied.  See the License for the
+       specific language governing permissions and limitations
+       under the License.
+*/
+
+package org.apache.cordova;
+
+public interface ICordovaCookieManager {
```

Comment:
The problem with using Android APIs and adopting them as our own is that we have to support stupid crap that isn't universal.  I'd rather add an API later that we needed than have some extra cruft that we have to test.  We should work on having the bare minimum needed for FileTransfer to work, and if a third-party API uses this, they can then submit a bug or send a pull request to add something to the API.  Adding to an API is an API change, but it's not a breaking one.